101-hub-and-spoke-sandbox/azuredeploy.json

```diff
@@ -0,0 +1,916 @@
+         "properties":{
+            "allowVirtualNetworkAccess":true,
+            "allowForwardedTraffic":true,
+            "allowGatewayTransit":false,
+            "useRemoteGateways":true,
+            "remoteVirtualNetwork":{
+               "id":"[variables('hubID')]"
+            }
+         }
+      },
+      {
+         "condition":"[equals(parameters('deployVpnGateway'), 'No')]",
+         "apiVersion":"2018-02-01",
+         "type":"Microsoft.Network/virtualNetworks/virtualNetworkPeerings",
+         "name":"[concat(variables('prodSpokeVnetName'), '/peering_', variables('prodSpokeVnetName'), '_', variables('hubVnetName'))]",
+         "location":"[resourceGroup().location]",
+         "dependsOn":[
+            "[variables('prodSpokeID')]",
+            "[variables('hubID')]"
+         ],
+         "properties":{
+            "allowVirtualNetworkAccess":true,
+            "allowForwardedTraffic":true,
+            "allowGatewayTransit":false,
+            "useRemoteGateways":false,
+            "remoteVirtualNetwork":{
+               "id":"[variables('hubID')]"
+            }
+         }
+      },
+      {
+         "comments":"Windows Jump-Hosts",
+         "type":"Microsoft.Compute/virtualMachines",
+         "name":"[variables('winJmphostName')]",
+         "apiVersion":"2017-12-01",
+         "location":"[resourceGroup().location]",
+         "scale":null,
+         "properties":{
+            "hardwareProfile":{
+               "vmSize":"[parameters('winVmSize')]"
+            },
+            "storageProfile":{
+               "imageReference":{
+                  "publisher":"MicrosoftWindowsServer",
+                  "offer":"WindowsServer",
+                  "sku":"[parameters('windowsOSVersion')]",
+                  "version":"latest"
+               },
+               "osDisk":{
+                  "createOption":"FromImage"
+               },
+               "dataDisks":[
+                  {
+                     "diskSizeGB":20,
+                     "lun":0,
+                     "createOption":"Empty"
+                  }
+               ]
+            },
+            "osProfile":{
+               "computerName":"[variables('winJmphostName')]",
+               "adminUsername":"[parameters('winVmUser')]",
+               "adminPassword":"[parameters('winVmPassword')]",
+               "windowsConfiguration":{
+                  "provisionVMAgent":true
+               },
+               "secrets":[
+
+               ]
+            },
+            "networkProfile":{
+               "networkInterfaces":[
+                  {
+                     "id":"[resourceId('Microsoft.Network/networkInterfaces', variables('winVmNicName'))]"
+                  }
+               ]
+            },
+            "diagnosticsProfile":{
+               "bootDiagnostics":{
+                  "enabled":true,
+                  "storageUri":"[concat('https', '://', variables('winVmStorageName'), '.blob.core.windows.net', '/')]"
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris